pkg/cmd/project/item-edit/item_edit.go

```diff
@@ -0,0 +1,286 @@
+	// format
+	format string
+}
+
+type editItemConfig struct {
+	tp     *tableprinter.TablePrinter
+	client *api.GraphQLClient
+	opts   editItemOpts
+}
+
+type EditProjectDraftIssue struct {
+	UpdateProjectV2DraftIssue struct {
+		DraftIssue queries.DraftIssue `graphql:"draftIssue"`
+	} `graphql:"updateProjectV2DraftIssue(input:$input)"`
+}
+
+type UpdateProjectV2FieldValue struct {
+	Update struct {
+		Item queries.ProjectItem `graphql:"projectV2Item"`
+	} `graphql:"updateProjectV2ItemFieldValue(input:$input)"`
+}
+
+func NewCmdEditItem(f *cmdutil.Factory, runF func(config editItemConfig) error) *cobra.Command {
+	opts := editItemOpts{}
+	editItemCmd := &cobra.Command{
+		Use:   "item-edit",
+		Short: "Edit an item in a project by ID",
+		Long: `
+Edit one of a draft issue or a project item. Both require the ID of the item to edit. For non-draft issues, the ID of the project is also required, and only a single field value can be updated per invocation. See the flags for more details.`,
+		Example: `
+# add --format=json to output in JSON format
+
+# edit a draft issue title and body
+gh project item-edit --id DRAFT_ISSUE_CONTENT_ID --title "a new title" --body "a new body"
+
+# edit an item's text field value
+gh project item-edit --id ITEM_ID --field-id FIELD_ID --project-id PROJECT_ID --text "new text"
+
+# edit an item's number field value
+gh project item-edit --id ITEM_ID --field-id FIELD_ID --project-id PROJECT_ID --number 1
+
+# edit an item' date field value
+gh project item-edit --id ITEM_ID --field-id FIELD_ID --project-id PROJECT_ID --date "2023-01-01"
+
+# edit an item's single-select field value
+# you can retrieve the option ID from the output of the 'field-list --format=json' command
+gh project item-edit --id ITEM_ID --field-id FIELD_ID --project-id PROJECT_ID --single-select-option-id OPTION_ID
+
+# edit an item's iteration field value
+gh project item-edit --id ITEM_ID --field-id FIELD_ID --project-id PROJECT_ID --iteration-id ITERATION_ID
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := queries.NewClient()
+			if err != nil {
+				return err
+			}
+
+			t := tableprinter.New(f.IOStreams)
+			config := editItemConfig{
+				tp:     t,
+				client: client,
+				opts:   opts,
+			}
+
+			// allow testing of the command without actually running it
+			if runF != nil {
+				return runF(config)
+			}
+			return runEditItem(config)
+		},
+	}
+
+	editItemCmd.Flags().StringVar(&opts.itemID, "id", "", "ID of the item to edit (required). For draft issues, the ID is for the draft issue content which is prefixed with `DI_`. For other issues, it is the ID of the project item.")
+	editItemCmd.Flags().StringVar(&opts.format, "format", "", "Output format, must be 'json'.")
+
+	editItemCmd.Flags().StringVar(&opts.title, "title", "", "DRAFT ISSUE - Title of the draft issue item to edit.")
+	editItemCmd.Flags().StringVar(&opts.body, "body", "", "DRAFT ISSUE - Body of the draft issue item to edit.")
```

Comment:
My mind boggles at this "draft issue" feature. Looking at its entire doc string and the flag set, it almost feels like `item-edit` is a bit overloaded. But short of changing that (I wouldn't suggest it at it point), could we do something to make the usage of this clearer? I'm afraid I'm not familiar with the "draft issue" feature of Projects.

Reply:
Draft Issues are items that exist only a project - they do not have their own URL like an Issue or a PR (which is created then added to a Project). However, they can have values such as a number or an assignee, in the same way that Issues or PRs on a Project can.

I agree that the flags for this command are a bit of a mess.  I had a conversation with @samcoe about it here https://github.com/github/gh-projects/issues/84 while I was working on it. I also tried out flag groups in cobra to try to group the draft flags and the non-draft flags, without success.

If you have any ideas for how to improve this, please let me know.